sdk/rust/src/pg.rs

```diff
@@ -0,0 +1,207 @@
+//! Conversions between Rust, WIT and **Postgres** types.
+//!
+//! # Types
+//!
+//! | Rust type  | WIT (db-value)      | Postgres type(s)             |
+//! |------------|---------------------|----------------------------- |
+//! | `bool`     | boolean(bool)       | BOOL                         |
+//! | `i16`      | int16(s16)          | SMALLINT, SMALLSERIAL, INT2  |
+//! | `i32`      | int32(s32)          | INT, SERIAL, INT4            |
+//! | `i64`      | int64(s64)          | BIGINT, BIGSERIAL, INT8      |
+//! | `f32`      | floating32(float32) | REAL, FLOAT4                 |
+//! | `f64`      | floating64(float64) | DOUBLE PRECISION, FLOAT8     |
+//! | `String`   | str(string)         | VARCHAR, CHAR(N), TEXT       |
+//! | `Vec<u8>`  | binary(list<u8>)    | BYTEA                        |
+
+wit_bindgen_rust::import!("../../wit/ephemeral/outbound-pg.wit");
+
+/// Exports the generated outbound Pg items.
+pub use outbound_pg::*;
+
+impl std::error::Error for PgError {}
+
+impl ::std::fmt::Display for PgError {
+    fn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
+        match self {
+            PgError::ConnectionFailed(err_msg)
+            | PgError::BadParameter(err_msg)
+            | PgError::QueryFailed(err_msg)
+            | PgError::ValueConversionFailed(err_msg)
+            | PgError::OtherError(err_msg) => write!(f, "Postgres error: {}", err_msg),
+            PgError::Success => panic!("Unexpected error: Success isn't supposed to be used"),
+        }
+    }
+}
+
+// A pg error
```

Comment:
Not sure if I'm misreading this but I think `pub` stuff should have doc strings?  The module `deny(missing-docs)` should be catching this though so perhaps I'm misunderstanding...

Reply:
SDK allows `missing_docs` https://github.com/fermyon/spin/blob/main/sdk/rust/src/lib.rs#L41-L60 because, otherwise it fails with:

```
error: missing documentation for an enum
  --> sdk/rust/src/lib.rs:54:5
   |
54 |     wit_bindgen_rust::import!("../../wit/ephemeral/spin-config.wit");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `wit_bindgen_rust::import` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not document `spin-sdk`
```

But you're right, this isn't the reason for not adding the docs :)
Fixed, thanks!